### PR TITLE
Acceptance tests fix

### PR DESCRIPTION
**Pull request by jdziurzynski**
KNTK-349
Acceptance tests workflow fix.

TF_ACC_PREFIX is now correctly assigned.
Additional deletion of resources created during tests.

### CODE REVIEW

**Review by mateuszmidor — DISMISSED** [review]

**Review by mateuszmidor — COMMENTED** [review]
I like the resources cleanup addition

**Review by mmac-m3a — CHANGES_REQUESTED** [review]
The segfault on name collision needs to be fixed. The test should fail cleanly reporting the error returned by the Kentik API.

So, instead of:
```
--- FAIL: TestAccDataSourceCloudExportItemAWS (0.87s)
    data_source_cloudexport_item_test.go:229: 
        	Error Trace:	data_source_cloudexport_item_test.go:229
        	Error:      	Received unexpected error:
        	            	client.CloudExports.Create: rpc error: code = Internal desc = internal error (name:DuplicateKeyError type:object) (errxid cats2xv7panhvcx3hb20)
        	Test:       	TestAccDataSourceCloudExportItemAWS
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x78 pc=0xbc3b9e]
```
it should report just:
```
rpc error: code = Internal desc = internal error (name:DuplicateKeyError type:object) (errxid cats2xv7panhvcx3hb20)
        	Test:       	TestAccDataSourceCloudExportItemAWS
```
(or something close to that).

Let' make the fix in this PR, while the problem is still easy to reproduce.

File: internal/provider/data_source_cloudexport_list_test.go [hunk 1]
```diff
@@ -56,7 +57,23 @@ func makeTestCloudExportDataSourceList(apiURL string) string {
 func TestAccDataSourceCloudExportList(t *testing.T) {
 	if skipIfNotAcceptance() {
 		checkRequiredEnvVariables(t)
-		assert.NoError(t, createTestAccCloudExportList())
+
+		ctx := context.Background()
+		client, err := newClient()
+		assert.NoError(t, err)
+
+		ceAWS, err := client.CloudExports.Create(ctx, newAWSCloudExportListForAccTest())
```
[the remark this answers is not in the document]
Comment (jdziurzynski) [reply internal/provider/data_source_cloudexport_list_test.go:65]: Problem with panic should not occur anymore. In this PR I have changed the code to break the test when CloudExport client returns any error.
I tried to reproduce a similar situation with DuplicateKeyError and this is what I got:
```
?       github.com/kentik/terraform-provider-kentik-cloudexport [no test files]
--- FAIL: TestAccDataSourceCloudExportItemAWS (1.93s)
    data_source_cloudexport_item_test.go:234: 
                Error Trace:    data_source_cloudexport_item_test.go:234
                Error:          Received unexpected error:
                                rpc error: code = Internal desc = internal error (name:DuplicateKeyError type:object) (errxid caxcykjp64h36pan8fqg)
                Test:           TestAccDataSourceCloudExportItemAWS
FAIL
FAIL    github.com/kentik/terraform-provider-kentik-cloudexport/internal/provider       25.982s
FAIL
```
The last acceptance test was run by pull_request_target action and I think it run tests on the master branch, thats why we got panic.

[the remark this answers is not in the document]
Comment (jdziurzynski) [reply internal/provider/data_source_cloudexport_list_test.go:65]: TestAccDataSourceCloudExportList checks the kentik-cloudexport_list data source method. For this test, I created only two simple resources to check if kentik-cloudexport_list returns a resource list.
More advanced tests for all providers can be found at data_source_cloudexport_item_test.go

**Pull request merged**